AK/TypeCasts.h

```diff
@@ -18,6 +19,8 @@ ALWAYS_INLINE bool is(InputType& input)
     if constexpr (requires { input.template fast_is<OutputType>(); }) {
         return input.template fast_is<OutputType>();
     }
+    if constexpr (SameAs<InputType, OutputType>)
+        return true;
```

Comment:
Wait, why are we calling `is<T>` on a `T` to begin with? :thonk:

Reply:
I believe I've found the code that caused this issue in the first place: 
https://github.com/LadybirdBrowser/ladybird/blob/40742d95e66c98857fa43ea1c6632dbe75ecb3b3/Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp#L232

Would it be better to keep this change, or do you think it would be better to add `static_assert(!SameAs<InputType, OutputType>);` to the top of the function instead?

Reply:
Feels like the nicer fix is to simply remove the redundant `verify_cast`, or no?

Reply:
Yes, we should remove that. My thought was that adding the static assert would make it more obvious where these errors are occurring if they happen in the future.

Reply:
Oh I see! Yeah, that's a good idea :)